### PR TITLE
Don't report RS0027 if overload is obsolete

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes https://github.com/dotnet/roslyn-analyzers/issues/4766.

~~I currently excluded obsolete overloads from RS0027 analysis only. Let me know if the same should be done for RS0026.~~ (Done.)

### CODE REVIEW

**Comment by mavasani** [comment]
Thanks @Youssef1313 - yes RS0026 should have similar check as well.

**Review by mavasani — APPROVED** [review]
LGTM. Will wait for you to address RS0026 before merging.

File: src/PublicApiAnalyzers/UnitTests/DeclarePublicApiAnalyzerTests.cs [hunk 1]
```diff
@@ -984,6 +984,33 @@ public class C
...
+        {
+            var source = @"
+using System;
+
+public class C
+{
+    public void M(int p1 = 0) { }
+
+    [Obsolete]
+    public void M(char p1, int p2) { }
```
Comment (genlu) [inline src/PublicApiAnalyzers/UnitTests/DeclarePublicApiAnalyzerTests.cs:998]: Could you add an overload to test RS0026 too?

Comment (Evangelink) [reply src/PublicApiAnalyzers/UnitTests/DeclarePublicApiAnalyzerTests.cs:998]: @genlu `RS0026` is currently not excluded. There is an open question from @Youssef1313 on this.

**Review by Evangelink — APPROVED** [review]

File: src/PublicApiAnalyzers/UnitTests/DeclarePublicApiAnalyzerTests.cs (hunk 1 above)
Comment (Youssef1313) [reply src/PublicApiAnalyzers/UnitTests/DeclarePublicApiAnalyzerTests.cs:998]: @Evangelink I excluded it in 3944643 based on https://github.com/dotnet/roslyn-analyzers/pull/4773#issuecomment-772635499.

This only needs a test. Will add it soon.

Comment (Evangelink) [reply src/PublicApiAnalyzers/UnitTests/DeclarePublicApiAnalyzerTests.cs:998]: My bad, sorry :)

**Comment by genlu** [comment]
@Youssef1313 Thanks!

**Pull request merged**